src/components/global-view-core.js

```diff
@@ -0,0 +1,434 @@
+                step.taskData &&
+                step.taskData.status.includes(
+                    localFilter ? localFilter : statusFilter
+                )
+            );
+        });
+    };
+
+    const getMissingData = useCallback(
+        (allSteps, consideredPage, consideredRowsPerPage) => {
+            let allPromise = [];
+            let searchDays = [];
+            for (
+                let i = consideredPage * consideredRowsPerPage;
+                i <
+                    consideredPage * consideredRowsPerPage +
+                        consideredRowsPerPage && i < allSteps.length;
+                i++
+            ) {
+                if (!allSteps[i].taskData) {
+                    setIsLoading(true);
+                    let day = new Date(allSteps[i].timestamp)
+                        .toISOString()
+                        .substring(0, 10);
+                    if (!searchDays.includes(day)) {
+                        allPromise.push(
+                            fetchBusinessDateData(
+                                new Date(allSteps[i].timestamp)
+                                    .toISOString()
+                                    .substring(0, 10),
+                                intlRef,
+                                enqueueSnackbar
+                            )
+                        );
+                        searchDays.push(day);
+                    }
+                }
+            }
+            Promise.all(allPromise).then((values) => {
+                values.forEach((tasksdata) => {
+                    tasksdata.forEach((td) => {
+                        let globalIndex = allSteps.findIndex(
+                            (step) =>
+                                Date.parse(td.timestamp) === step.timestamp ||
+                                td.timestamp === step.timestamp
+                        );
+                        if (globalIndex >= 0)
+                            allSteps[globalIndex].taskData = td;
+                    });
+                });
+                setIsLoading(false);
+                setSteps(allSteps);
+                setStatusFilter('');
+            });
+        },
+        [enqueueSnackbar, intlRef]
+    );
+
+    useEffect(() => {
+        let newSteps = createAllSteps(
+            timestampMin,
+            timestampMax,
+            timestampStep
+        );
+        setPage(0);
+        setOpenEvent([]);
+        getMissingData(newSteps, 0, rowsPerPage);
+    }, [
+        timestampMin,
+        timestampMax,
+        timestampStep,
+        rowsPerPage,
+        getMissingData,
+    ]);
+
+    const getEventsData = () => {
+        let index = openEvent.indexOf(true);
+        return index >= 0 ? steps[index].taskData : [];
+    };
+
+    const getFilesData = (field) => {
+        let index = openEvent.indexOf(true);
+        return index >= 0 ? steps[index].taskData[field] : [];
+    };
+
+    const inputDataRow = (step, index) => {
```

Comment:
Maybe extract that as another component

Reply:
The new component GlobalViewCoreRow has been created